src/Controllers/Post/PostController.php

```diff
@@ -27,11 +27,17 @@ public function execute(int $id)
             }
         }
         $post = $this->postRepository->getPost($id);
-        $comments = $this->commentRepository->getComments($id);
+        $comments = $this->commentRepository->getPostComments($id);
+        $approvedComments = [];
+        for ($i = 0; $i < count($comments); $i++) {
```

Comment:
foreach seems more appropriate or what about a method that only retrieve approved comments from the repository
$this->commentRepository->getApprovedComments($postId);